Mathlib/NumberTheory/NumberField/Embeddings.lean

```diff
@@ -603,6 +606,21 @@ theorem card_add_two_mul_card_eq_rank :
 
 variable {K}
 
+theorem nrComplexPlaces_eq_zero_of_finrank_eq_one (h : finrank ℚ K = 1) :
+    NrComplexPlaces K = 0 := by
+  have := card_add_two_mul_card_eq_rank K
+  rw [h, Nat.add_eq_one_iff] at this
+  rcases this with (⟨-, H⟩ | ⟨-, H⟩)
+  · exfalso
+    exact (Nat.even_iff_not_odd.1 <| (even_iff_exists_two_mul 1).2 ⟨NrComplexPlaces K, H.symm⟩)
+      ⟨0, rfl⟩
+  · simpa using H
```

Comment:
```suggestion
  linarith
```

Reply:
Wow, `linarith` is smater than I thought!